lib/src/mock_document_reference.dart

```diff
@@ -147,13 +147,24 @@ class MockDocumentReference<T extends Object?> implements DocumentReference<T> {
       final fieldValuePlatform = valueDelegate as MockFieldValuePlatform;
       final fieldValue = fieldValuePlatform.value;
       fieldValue.updateDocument(document, key);
-    } else if (value is DateTime) {
-      document[key] = Timestamp.fromDate(value);
     } else {
-      document[key] = value;
+      document[key] = _transformValue(value);
     }
   }
 
+  dynamic _transformValue(dynamic value) {
```

Comment:
`_transformValue` is quite a vague name, so another developer wouldn't know what it does at first glance. Also, it is the same as `_transformDates` in the other file, except this one supports Maps as well. Let's make it a reusable function instead of duplicating code.

Would you mind moving this function to https://github.com/atn832/fake_cloud_firestore/blob/master/lib/src/util.dart, renaming it to `_transformDates` and document it to explain that it will also recursively convert `DateTime` to `Timestamp` in nested structures such as maps and lists?

Then you can have both mock_document_reference.dart and mock_query.dart use it.

Reply:
Good catch, I overlooked this file.